### PR TITLE
added more assertions

### PR DESCRIPTION
**Pull request by profes**
fixes https://github.com/ensime/ensime-server/issues/1372

### CODE REVIEW

**Comment by profes** [comment]
@fommil build failed... any reason why?

**Comment by rorygraves** [comment]
@profes Looks like a emacs tests failure (we have a few transient failures lurking which we are trying to kill (and fewer since @fommil 's last attack)) - Will wait for @fommil to confirm, but looks good to me.

**Comment by fommil** [comment]
Yeah yeah. I should write a template reply explaining that I broke the build, I'm wearing the naughty hat, etc etc

**Comment by fommil** [comment]
Btw, good to see you taking up lisp @rorygraves

**Comment by rorygraves** [comment]
My comment was more you tend to fix the build...

**Comment by rorygraves** [comment]
lol - no lisp here - its a regex - 'oh it failed in the emacs tests' ;)

**Comment by fommil** [comment]
can you please rebase?

**Comment by profes** [comment]
done

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,4 @@ target/
 # RELATED TO THE BUILD.
 #
 # If you want to add something from your personal setup,
-# add it to your ~/.gitignore instead.
+# add it to your ~/.gitignore instead.
```
Comment (fommil) [inline .gitignore]: this is a whitespace change

**Comment by fommil** [comment]
I don't know why the Windows tests failed, I'll have to look into that when I get home.

Can you please remove the newline change to `.gitignore`?

**Comment by profes** [comment]
fixed, my bad ;)

**Comment by fommil** [comment]
Thank you!

**Pull request merged**